examples/versatileab/src/bin/prefetch-exception.rs

```diff
@@ -0,0 +1,53 @@
+//! Example triggering an prefetch exception.
+
+#![no_std]
+#![no_main]
+
+use cortex_ar::register::{Ifar, Ifsr};
+// pull in our start-up code
+use versatileab as _;
+
+use semihosting::println;
+
+versatileab::entry_point!();
+
+/// The main function of our Rust application.
+#[export_name = "main"]
+fn main() -> ! {
+    println!("Hello, this is an prefetch exception example");
+
+    // A BKPT instruction triggers a Prefetch Abort except when Halting debug-mode is enabled.
+    // See p. 2038 of ARMv7-M Architecture Reference Manual
+    unsafe {
+        core::arch::asm!("bkpt");
+    }
+
+    unreachable!("should never be here!");
+}
+
+#[unsafe(no_mangle)]
+unsafe extern "C" fn _undefined_handler(_faulting_instruction: u32) {
+    panic!("unexpected undefied exception");
+}
+
+#[unsafe(no_mangle)]
+unsafe extern "C" fn _prefetch_handler(_faulting_instruction: u32) {
+    static mut COUNTER: u32 = 0;
```

Comment:
Re-entering this function using a static mut is UB. It should probably be at AtomicU32 using fetch_add.